crypto/evp/evp_enc.c

```diff
@@ -1675,6 +1879,7 @@ static void *evp_cipher_from_algorithm(const int name_id,
     }
     if ((fnciphcnt != 0 && fnciphcnt != 3 && fnciphcnt != 4)
             || (fnciphcnt == 0 && cipher->ccipher == NULL)
+            || (fnpipecnt != 0 && fnpipecnt != 3 && fnpipecnt != 4)
```

Comment:
This doesn't seem quite right since it would allow no final as long as you had p_einit, p_dinit and p_cupdate.

Reply:
I overlooked it. Nonetheless, this pertains to non-pipeline variants as well. As I do not want to modify existing behavior, I will be fixing only the pipeline count.